src/code/utils/codap-connect.coffee

```diff
@@ -0,0 +1,49 @@
+IframePhoneRpcEndpoint = require '../../vendor/iframe-phone-rpc-endpoint'
+
+module.exports = class CodapConnect
+
+  name: 'Building Models Tool'
+
+  codapPhone: null
+
+  initAccomplished: false
+
+  init: (linkManager, name) ->
+    log.info 'CodapConnect initializing'
+    @linkManager = linkManager
+    name and @name = name
+
+    @codapPhone = new IframePhoneRpcEndpoint( @codapRequestHandler,
+      'codap-game', window.parent )
+
+    @codapPhone.call( {
+      action: 'initGame'
+      args:
+        name: @name
+        dimensions:
+          width: 800
+          height: 600
+        contextType: 'DG.DataContext'
+    }, @initGameHandler)
+
+  codapRequestHandler: (iCmd, iCallback) =>
```

Comment:
No, the [current API](https://github.com/concord-consortium/codap/wiki/Data-Interactive-API) is mostly initiated from the Data Interactive. Only the save and restore requests come from the CODAP side. 

Some existing Data Interactives implement helper functions to wrap the calls to codap, others build directly on the IFramePhone API. I am not sure what is appropriate here.